src/main/java/dev/slimevr/osc/VRCOSCHandler.java

```diff
@@ -0,0 +1,252 @@
+			if (shareableTrackers.get(i).getTrackerRole() != TrackerRole.HEAD) {
+				trackersEnabled[i] = config
+					.getOSCTrackerRole(shareableTrackers.get(i).getTrackerRole(), false);
+			}
+		}
+
+		// Stops listening and closes OSC port
+		if (oscReceiver != null && oscReceiver.isListening()) {
+			oscReceiver.stopListening();
+		}
+		if (oscSender != null && oscSender.isConnected()) {
+			try {
+				oscSender.close();
+			} catch (IOException e) {
+				LogManager.warning("[VRCOSCHandler] Error closing the OSC sender: " + e);
+			}
+		}
+
+		if (config.getEnabled()) {
+			// Instantiates the OSC receiver
+			try {
+				int port = config.getPortIn();
+				oscReceiver = new OSCPortIn(
+					port
+				);
+				LogManager.info("[VRCOSCHandler] Listening to port " + port);
+			} catch (IOException e) {
+				LogManager
+					.warning("[VRCOSCHandler] Error listening to the port " + config.getPortIn());
+			}
+
+			// Starts listening for the Upright parameter from VRC
+			OSCMessageListener listener = this::handleReceivedMessage;
+			MessageSelector selector = new OSCPatternAddressMessageSelector(
+				"/avatar/parameters/Upright"
+			);
+			oscReceiver.getDispatcher().addListener(selector, listener);
+			oscReceiver.startListening();
+
+			// Instantiate the OSC sender
+			try {
+				InetAddress address = config.getAddress();
+				int port = config.getPortOut();
+				oscSender = new OSCPortOut(
+					address,
+					port
+				);
+				oscSender.connect();
+				LogManager
+					.info(
+						"[VRCOSCHandler] Sending to port "
+							+ port
+							+ " at address "
+							+ address.toString()
+					);
+			} catch (IOException e) {
+				LogManager
+					.warning(
+						"[VRCOSCHandler] Error connecting to port "
+							+ config.getPortOut()
+							+ " at the address "
+							+ config.getAddress()
+					);
+			}
+		}
+	}
+
+	void handleReceivedMessage(OSCMessageEvent event) {
+		timeAtLastOSCMessageReceived = System.currentTimeMillis();
+		if (steamvrBridge != null && steamvrBridge.isConnected()) {
+			if (hmd.getStatus() != TrackerStatus.OK) {
+				hmd.setStatus(TrackerStatus.OK);
+			}
+			hmd.position
+				.set(
+					0f,
+					(float) event
+						.getMessage()
+						.getArguments()
+						.get(0) * humanPoseProcessor.getUserHeightFromConfig(),
+					0f
+				);
+			hmd.rotation.set(Quaternion.IDENTITY);
+			hmd.dataTick();
+		}
+	}
+
+	public void update() {
+		// Manage HMD state with timeout
+		if (oscReceiver != null) {
+			if (
+				((steamvrBridge != null
+					&& steamvrBridge.isConnected())
+					||
+					System.currentTimeMillis() - timeAtLastOSCMessageReceived > HMD_TIMEOUT
+					||
+					!oscReceiver.isListening())
+					&& hmd.getStatus() == TrackerStatus.OK
+			) {
+				hmd.setStatus(TrackerStatus.DISCONNECTED);
+			}
+		}
+
+		// Send OSC data
+		if (oscSender != null && oscSender.isConnected()) {
+			for (int i = 0; i < shareableTrackers.size(); i++) {
+				if (trackersEnabled[i]) {
```

Comment:
This line is scary, but looking into the code for how `shareableTrackers` is acquired, there's not much that can be done about it... A check for the bounds of `trackersEnabled` is possible but really not necessary, I *really* don't like how `HumanPoseProcessor` does this

Reply:
I'm gonna resolve this due to its irrelevance to this PR, hopefully it can just be tackled in some refactor later down the line